extensions/wikia/SitemapXml/SpecialSitemap_body.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/**
+ * Main part of Special:Sitemap
+ *
+ * @file
+ * @ingroup Extensions
+ * @author Krzysztof Krzyżaniak <eloy@wikia-inc.com> for Wikia Inc.
+ * @copyright © 2010, Wikia Inc.
+ * @license http://www.gnu.org/copyleft/gpl.html GNU General Public License 2.0 or later
+ * @version 1.0
+ */
+
+if ( !defined( 'MEDIAWIKI' ) ) {
+	echo "This is a MediaWiki extension and cannot be used standalone.\n";
+	exit( 1 );
+}
+
+class SitemapPage extends UnlistedSpecialPage {
+
+	/**
+	 * standard constructor
+	 * @access public
+	 * @param $name
+	 */
+	public function __construct( $name = 'Sitemap' ) {
+		parent::__construct( $name );
+	}
+
+	/**
+	 * Main entry point
+	 *
+	 * @access public
+	 *
+	 * @param $subpage Mixed: subpage of SpecialPage
+	 */
+	public function execute( $subpage ) {
+		global $wgOut;
+
+		// keeping the checks for old sitemaps for now so we return 404s
+		$showIndex = strpos( $subpage, '-index.xml' ) !== false;
+
+		$forceOldSitemap = strpos( $subpage, '-oldsitemapxml-' ) !== false;
+		$forceNewSitemap = strpos( $subpage, '-newsitemapxml-' ) !== false;
+
+		if ( ( $showIndex && !$forceOldSitemap ) || $forceNewSitemap ) {
+			$wgOut->disable();
```

Comment:
We can avoid the global with `$this->getOutput()->disable();`

Reply:
Done. In my defence, I just moved this file :)